frontend/src/app/core/services/auth.service.ts

```diff
@@ -26,11 +27,18 @@ export class AuthService {
     ) {}
 
     signIn(email: string, password: string) {
-        return this.afAuth.signInWithEmailAndPassword(email, password).then((userCredential) => {
-            if (userCredential.user) {
-                this.setAccessToken(userCredential.user);
-            }
-        });
+        const firebasePromise = this.afAuth.signInWithEmailAndPassword(email, password);
+
+        const corePromise = firstValueFrom(this.userService.getUser().pipe(catchError(() => {
+            throw new Error('User was incorrectly registered, please try another one');
+        })));
```

Comment:
You won't be able to get user without firebase response, aren't you?
So, you can't send this requests in parallel